jetty-websocket/websocket-core-common/src/main/java/org/eclipse/jetty/websocket/core/internal/PerMessageDeflateExtension.java

```diff
@@ -146,6 +147,17 @@ public void init(final ExtensionConfig config, WebSocketComponents components)
         super.init(configNegotiated, components);
     }
 
+    @Override
+    public void close()
+    {
+        // TODO: use IteratingCallback.close() instead of creating exception with failFlusher methods.
+        ClosedChannelException exception = new ClosedChannelException();
```

Comment:
You should probably use the same `fillInStackTrace` trick as above, otherwise this `close()` implementation is going to be very expensive.

Reply:
Is there a reason close doesn't work now in this PR?

Reply:
Close doesn't work now because `DemandingFlusher` and `TransformingFlusher` will still allow new entries to be added and may not fail existing entries if close is called. I think we should carefully assess this in a separate PR.

Reply:
Opened Issue #8137

Reply:
Ouch! That sounds like a bad bug in close... so please make 8137 a moderate high priority.